keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java

```diff
@@ -265,9 +263,34 @@ private void closeClients() {
     closeTestSecretsManager();
   }
 
-  private String createSecret(final Tag tag, String secretValue)
+  private boolean checkTestSecretExists(
+      final String testSecretName, final Tag tag, final String secretValue) {
+    try {
+      final GetSecretValueResponse getSecretValueResponse =
+          testSecretsManagerClient
+              .getSecretValue(GetSecretValueRequest.builder().secretId(testSecretName).build())
+              .get(30, TimeUnit.SECONDS);
+      final DescribeSecretResponse describeSecretResponse =
+          testSecretsManagerClient
+              .describeSecret(DescribeSecretRequest.builder().secretId(testSecretName).build())
+              .get(30, TimeUnit.SECONDS);
+      return getSecretValueResponse.name().equals(testSecretName)
+          && getSecretValueResponse.secretString().equals(secretValue)
+          && describeSecretResponse.hasTags()
+          && describeSecretResponse.tags().get(0).key().equals(tag.key())
+          && describeSecretResponse.tags().get(0).value().equals(tag.value());
```

Comment:
```suggestion
return getSecretValueResponse.secretString().equals(secretValue)
          && describeSecretResponse.tags().contains(tag);
```

You used the testSecretName to get fetch the details so I think it's redundant to check its equality here too.
Can also use contains and rely on Tag's .equals()